tests/test_qnn.py

```diff
@@ -0,0 +1,325 @@
+            return kwargs
+
+        weight_specs = {
+            "w1": {"initializer": "random_uniform", "trainable": False},
+            "w2": {"initializer": RandomNormal(mean=0, stddev=0.5)},
+            "w3": {},
+            "w4": {},
+            "w5": {},
+        }
+
+        with monkeypatch.context() as m:
+            m.setattr(Layer, 'add_weight', add_weight_dummy)
+            c, w = get_circuit
+            layer = KerasLayer(c, w, output_dim, weight_specs=weight_specs)
+            layer.build(input_shape=(10, n_qubits))
+
+            for weight in layer.weight_shapes:
+                assert all(item in layer.qnode_weights[weight].items() for item in weight_specs[
+                    weight].items())
+
+    @pytest.mark.parametrize("n_qubits, output_dim", indices(3))
+    @pytest.mark.parametrize("input_shape", [(10, 4), (8, 3)])
+    def test_compute_output_shape(self, get_circuit, output_dim, input_shape):
+        """Test if the compute_output_shape() method performs correctly, i.e., that it replaces
+        the last element in the input_shape tuple with the specified output_dim and that the
+        output shape is of type tf.TensorShape"""
+        c, w = get_circuit
+        layer = KerasLayer(c, w, output_dim)
+
+        assert layer.compute_output_shape(input_shape) == (input_shape[0], output_dim)
+        assert isinstance(layer.compute_output_shape(input_shape), tf.TensorShape)
+
+    @pytest.mark.parametrize("n_qubits, output_dim", indices(4))
+    @pytest.mark.parametrize("batch_size", [5, 10, 15])
+    def test_call(self, get_circuit, output_dim, batch_size, n_qubits):
+        """Test if the call() method performs correctly, i.e., that it outputs with shape
+        (batch_size, output_dim) with results that agree with directly calling the QNode"""
+        c, w = get_circuit
+        layer = KerasLayer(c, w, output_dim)
+        x = tf.ones((batch_size, n_qubits))
+
+        layer_out = layer(x)
+        weights = [w[0] if w.shape == (1,) else w for w in layer.qnode_weights.values()]
+
+        assert layer_out.shape == (batch_size, output_dim)
+        assert np.allclose(layer_out[0], c(*weights, x=x[0]))
+
+    @pytest.mark.parametrize("n_qubits, output_dim", indices(1))
+    def test_str_repr(self, get_circuit, output_dim):
+        """Test the __str__ and __repr__ representations"""
+        c, w = get_circuit
+        layer = KerasLayer(c, w, output_dim)
+
+        assert layer.__str__() == "<Quantum Keras layer: func=circuit>"
+        assert layer.__repr__() == "<Quantum Keras layer: func=circuit>"
+
+
+tf.keras.backend.set_floatx('float64')  # TODO fix
+
+
+@pytest.mark.usefixtures("get_circuit", "model")
+class TestKerasLayerIntegration:
+    """Integration tests for the pennylane.qnn.KerasLayer class."""
+
+    @pytest.mark.parametrize("n_qubits, output_dim", indices(2))
+    @pytest.mark.parametrize("batch_size", [5, 10])
+    def test_train_model(self, model, batch_size, n_qubits, output_dim):
```

Comment:
I agree; I think it's sufficient to just check that an update step works without error, independent of what the new loss may be.

Reply:
Thanks @co9olguy @josh146, I wasn't happy with the `assert` in this test.

Is it enough to have a test without an assert, just to check if something works without raising an exception? I think this is the best we can do here.

Reply:
> Is it enough to have a test without an assert, just to check if something works without raising an exception?

I believe so, yes, even though it does make me uneasy. I think it's because I'm so used to every test having an assert statement.